core/dbt/config/project.py

```diff
@@ -835,10 +837,18 @@ def read_project_flags(project_dir: str, profiles_dir: str) -> ProjectFlags:
             project_flags = profile_project_flags
 
         if project_flags is not None:
+            # if warn_error_options are set, handle collapsing `include` and `error` as well as
+            # collapsing `exclude` and `warn`
+            warn_error_options = project_flags.get("warn_error_options")
+            if warn_error_options:
```

Comment:
non-blocking suggestion: It could be worth having `exclusive_primary_alt_value_setting` handle `None` as an input so that callers don't have to concern themselves with this logic, given that the preprocessing of `warn_error_options` is happening outside the `WarnErrorOptions` class in dbt-common and may need to be duplicated